src/main/java/org/magicdgs/readtools/utils/trimming/TrimAndFilterPipeline.java

```diff
@@ -271,31 +268,15 @@ public boolean test(final GATKRead read) {
      * @throws CommandLineException.BadArgumentValue if no trimmer and filter instances are
      *                                               specified.
      */
-    // TODO: change the signature once the new version of Barclay includes the method to get defaults
     public static TrimAndFilterPipeline fromPluginDescriptors(
-            final TrimmerPluginDescriptor trimmingPlugin,
-            final GATKReadFilterPluginDescriptor filterPlugin) {
+            final CommandLinePluginDescriptor<TrimmingFunction> trimmingPlugin,
+            final CommandLinePluginDescriptor<ReadFilter> filterPlugin) {
 
         // add the default and afterwards the ones provided by the user
-        final List<TrimmingFunction> trimmers = trimmingPlugin.getDefaultInstances().stream()
-                .map(tf -> (TrimmingFunction) tf) // TODO: this is necessary because it is now a set of filters
-                .collect(Collectors.toList());
-        trimmers.addAll(trimmingPlugin.getAllInstances());
-
-        // TODO: change to the same for the ReadFilter once the Barclay interface is fixed
-        // TODO: https://github.com/broadinstitute/barclay/pull/38
-        // the same for filters
-//        final List<ReadFilter> filters = filterPlugin.getDefaultInstances().stream()
-//                .map(rf -> (ReadFilter) rf) // TODO: this is necessary because it is now a set of filters
-//                .collect(Collectors.toList());
-//        filters.addAll(filterPlugin.getAllInstances());
+        final List<TrimmingFunction> trimmers = trimmingPlugin.getResolvedInstances();
 
         // using the getMergedReadFilter is hack, because it does not return any merged filter
-        final List<ReadFilter> filters = new ArrayList<>(filterPlugin.getAllInstances().size());
-        filterPlugin.getMergedReadFilter(new SAMFileHeader(), (finalFilters, header) -> {
-                    filters.addAll(finalFilters);
-                    return null;
-        });
+        final List<ReadFilter> filters = filterPlugin.getResolvedInstances();
```

Comment:
I believe that this solves https://github.com/magicDGS/ReadTools/issues/180, so should be closed too

Reply:
True (edit the issue)